src/cmdliner_arg.ml

```diff
@@ -124,11 +124,22 @@ let flag_all a =
   arg_to_args a, convert
 
 let vflag v l =
-  let convert _ cl =
+  let convert ei cl =
     let rec aux fv = function
     | (v, a) :: rest ->
         begin match Cmdliner_cline.opt_arg cl a with
-        | [] -> aux fv rest
+        | [] ->
+          begin match try_env ei a Cmdliner_base.env_bool_parse ~absent:false with
+          | Ok false ->
+            (* Env variable being false is equivalent to being absent *)
+            aux fv rest
+          | Ok true ->
+            let opt_name = Cmdliner_info.Arg.opt_name_sample a in
+            begin match fv with
+            | None -> aux (Some (Cmdliner_info.Arg.opt_name_sample a, v)) rest
+            | Some (g, _) -> failwith (Cmdliner_msg.err_opt_repeated g opt_name)
```

Comment:
> So this adds one env variable per case in the list. I'm not sure I'm very happy about the idea. This makes a lot of irrepresentable states legal

Similar to handling of the `--flag` options, which prevent duplication of arguments:
https://github.com/dbuenzli/cmdliner/blob/master/src/cmdliner_arg.ml#L135

The environment variables also prevent duplicated options